tests/test_pagination.py

```diff
@@ -703,6 +627,145 @@ def test_cursor_pagination(self):
         assert isinstance(self.pagination.to_html(), type(''))
 
 
+class TestCursorPagination(CursorPaginationTestsMixin):
```

Comment:
The large change footprint on these tests makes them v difficult to review. Is there any way around we could do this that ended up with a smaller review to judge this on?